src/components/run-button.js

```diff
@@ -121,27 +144,32 @@ const RunButton = (props) => {
         return props.getStatus(getRunnable());
     }
 
-    const runningStatus = getRunningStatus();
-
     let disabled =
-        (selectedIndex === 0 && runningStatus !== RunningStatus.IDLE) ||
-        (selectedIndex === 1 && runningStatus === RunningStatus.RUNNING);
+        (selectedIndex === 0 && getRunningStatus() !== RunningStatus.IDLE) ||
+        (selectedIndex === 1 && isRunning());
 
     return (
         <SplitButton
-            options={props.runnables}
+            options={getOptions(
+                getRunningStatus(),
+                props.runnables,
+                props.actionsOnRunnable
+            )}
             selectedIndex={selectedIndex}
             onSelectionChange={(index) => setSelectedIndex(index)}
             onClick={handleClick}
-            className={getStyle(runningStatus)}
+            className={getStyle(getRunningStatus())}
             buttonDisabled={disabled}
-            selectionDisabled={runningStatus === RunningStatus.RUNNING}
+            selectionDisabled={selectedIndex === 0 && isRunning()}
             startIcon={props.getStartIcon(getRunningStatus())}
             text={
                 props.getText
                     ? props.getText(getRunnable(), getRunningStatus())
                     : ''
             }
+            onStopComputation={handleStopComputationClick}
```

Comment:
I think it's weird that runnableActions are generic but in contrast onStopComputation is not. Either make everything generic or everything specific

Reply:
why is onStopComputation  not generic ?

Reply:
Done

Reply:
if you add a new runnableActions, for exemple "pause computation" or "restart computation", the onStopComputation callback is called (obviously you can't know what to call based only on the i18n label)